lib/benchfella.ex

```diff
@@ -96,12 +96,27 @@ defmodule Benchfella do
     print_results(results, bench_time, format, outdir, mem_stats, sys_mem_stats)
   end
 
-  defp prepare_tests_for_running(table) do
-    :ets.tab2list(table) |> Enum.group_by(fn {{mod, _test}} -> mod end)
+  defp prepare_tests_for_running(table, opts) do
+    rows = :ets.tab2list(table)
+    all = case opts[:line] do
+            nil ->
+              rows
+            line ->
+              # find closest test
+              first =
+                rows
+                |> Enum.sort(fn({{_, _, line_1}}, {{_, _, line_2}}) ->
+                 abs(opts[:line] - line_1) < abs(opts[:line] - line_2)
+                end)
+                |> List.first()
+              [first]
+          end
+    Enum.group_by(all, fn {{mod, _test, _line}} -> mod end)
   end
 
   # TODO: extract logging from this function and number running tests externally
   defp run_grouped_tests(groups, count, follow, bench_config) do
+    count = groups |> Map.values() |> List.flatten() |> Enum.count()
```

Comment:
Why does this calculation need to happen here? The function parameter `count` is no longer used at all.